infrastructure/src/main/java/org/corfudb/infrastructure/InvokeCheckpointingJvm.java

```diff
@@ -0,0 +1,98 @@
+package org.corfudb.infrastructure;
+
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class InvokeCheckpointingJvm implements IInvokeCheckpointing {
+
+    private static final long CONN_RETRY_DELAY_MILLISEC = 500;
+    private static final int MAX_COMPACTION_RETRIES = 8;
+    private final ServerContext serverContext;
+    private volatile Process checkpointerProcess;
+    private Logger syslog;
+    private boolean isInvoked;
+
+    public InvokeCheckpointingJvm(ServerContext serverContext) {
+        this.serverContext = serverContext;
+        syslog = LoggerFactory.getLogger("syslog");
+    }
+
+    @Override
+    public void invokeCheckpointing() {
+        for (int i = 1; i <= MAX_COMPACTION_RETRIES; i++) {
+            try {
+                if (!serverContext.getCompactorScriptPath().isPresent() || !serverContext.getCompactorConfig().isPresent()) {
+                    syslog.warn("Compactor client runner script or config file not found");
+                    return;
+                }
+
+                String compactorScriptPath = serverContext.getCompactorScriptPath().get();
+                String compactorConfigPath = serverContext.getCompactorConfig().get();
+                List<String> endpoint = Arrays.asList(serverContext.getLocalEndpoint().split(":"));
+                String hostName = endpoint.get(0);
+                String port = endpoint.get(1);
+
+                if (this.checkpointerProcess != null && this.checkpointerProcess.isAlive()) {
+                    this.checkpointerProcess.destroy();
```

Comment:
Functionality will not be affected. But invoking another redundant jvm would be unnecessary and it might even create extra load on the VM it's running.